tpu-provisioner/internal/controller/deletion_controller.go

```diff
@@ -147,6 +149,7 @@ func (r *DeletionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Node{}).
+		Watches(&corev1.Pod{}, handler.EnqueueRequestsFromMapFunc(handler.MapFunc(nodeForPod))).
```

Comment:
@Ethanlm Here is where we add the Pod watch back in:

`Pod terminates (we watch pod deletion event) --> We map Pod to its Node --> We reconcile on that Node --> We find the node pool for that node and delete if needed`